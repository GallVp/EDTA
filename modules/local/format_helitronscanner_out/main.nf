process FORMAT_HELITRONSCANNER_OUT {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/c0/c0905ae7aa2a1c4135b448b49dd205b4eb370d4733b832594563ee06832ebd09/data':
        'community.wave.seqera.io/library/perl:67db1a8d53d64b14' }"

    input:
    tuple val(meta), path(genome)
    path hel_fa
    path rc_hel_fa

    output:
    tuple val(meta), path("*.HelitronScanner.filtered.tabout")  , emit: filtered_tabout
    tuple val(meta), path("*.HelitronScanner.filtered.fa")      , emit: filtered_fa     , optional: true
    tuple val(meta), path("*.HelitronScanner.filtered.ext.fa")  , emit: filtered_ext_fa , optional: true
    tuple val(meta), path("*.format_helitronscanner_out.log")   , emit: log
    path "versions.yml"                                         , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix  = task.ext.prefix ?: "$genome"
    def args    = task.ext.args ?: ''
    """
    # Create symbolic links to match expected filenames
    ln -s $hel_fa ${genome}.HelitronScanner.draw.hel.fa
    ln -s $rc_hel_fa ${genome}.HelitronScanner.draw.rc.hel.fa

    format_helitronscanner_out.pl \\
        -genome $genome \\
        $args \\
        &> >(tee "${prefix}.format_helitronscanner_out.log" 2>&1)

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        perl: \$(perl -v | sed -n 's|This is perl.*(\\(.*\\)).*|\\1|p')
    END_VERSIONS
    """

    stub:
    def prefix  = task.ext.prefix ?: "$genome"
    """
    touch ${prefix}.HelitronScanner.filtered.tabout
    touch ${prefix}.format_helitronscanner_out.log

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        perl: \$(perl -v | sed -n 's|This is perl.*(\\(.*\\)).*|\\1|p')
    END_VERSIONS
    """
}
