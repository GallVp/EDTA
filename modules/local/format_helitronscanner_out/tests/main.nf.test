nextflow_process {

    name "Test Process FORMAT_HELITRONSCANNER_OUT"
    script "../main.nf"
    process "FORMAT_HELITRONSCANNER_OUT"

    tag "modules"
    tag "modules_local"
    tag "format_helitronscanner_out"

    test("sarscov2 - genome_fasta - stub") {

        options '-stub'

        when {
            process {
                """
                input[0] = [
                    [ id:'sarscov2' ],
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.fasta', checkIfExists: true)
                ]
                input[1] = []
                input[2] = []
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
