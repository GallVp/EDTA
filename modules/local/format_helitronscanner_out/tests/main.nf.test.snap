{
    "sarscov2 - genome_fasta - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "sarscov2"
                        },
                        "genome.fasta.HelitronScanner.filtered.tabout:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    [
                        {
                            "id": "sarscov2"
                        },
                        "genome.fasta.format_helitronscanner_out.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    "versions.yml:md5,250d1a5d5f60868f6cf0f17bd52edd91"
                ],
                "filtered_ext_fa": [
                    
                ],
                "filtered_fa": [
                    
                ],
                "filtered_tabout": [
                    [
                        {
                            "id": "sarscov2"
                        },
                        "genome.fasta.HelitronScanner.filtered.tabout:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "sarscov2"
                        },
                        "genome.fasta.format_helitronscanner_out.log:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,250d1a5d5f60868f6cf0f17bd52edd91"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-12-03T14:26:36.313536"
    }
}